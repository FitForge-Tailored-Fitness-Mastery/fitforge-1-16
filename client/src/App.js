// App.js
import React from 'react';
import { BrowserRouter as Router, Route, Routes, Navigate } from 'react-router-dom';
import Profile from './components/Profile/Profile';
import TrainerProfile from './components/TrainerProfile/TrainerProfile';
import EditProfile from './components/Profile/EditProfile'
import Login from './components/Login/Login';
import Signup from './components/Signup/Signup';
import Index from './components/Index/Index';
import ClientHome from './components/ClientHome/ClientHome';
import WorkoutHistory from './components/WorkoutHistory/WorkoutHistory';
import './App.css';


const PrivateRoute = ({ children }) => {
  const clientId = localStorage.getItem('clientId');
  
  return clientId ? children : <Navigate to="/" />;
}

const App = () => {
  return (
    <Router>
      <div className="App">
        <Routes>
          <Route path="/" element={<Login />} />
          <Route path="/signup" element={<Signup />} />
          <Route path="/index" element={<PrivateRoute><Index /></PrivateRoute>} />
          <Route path="/profile" element={<PrivateRoute><Profile /></PrivateRoute>} />
          <Route path="/trainerprofile/:clientId" element={<PrivateRoute><TrainerProfile /></PrivateRoute>} />
          <Route path="/edit-profile/:clientId" element={<PrivateRoute><EditProfile /></PrivateRoute>} />
          <Route path="/client-home" element={<PrivateRoute><ClientHome /></PrivateRoute>} />
          <Route path="/workout-history" element={<PrivateRoute><WorkoutHistory /></PrivateRoute>} />
        </Routes>
      </div>
    </Router>
  );
};

export default App;