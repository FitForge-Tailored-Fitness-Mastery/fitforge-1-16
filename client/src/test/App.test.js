import renderer from 'react-test-renderer';
import React from 'react';
import App from '../App';

it('testing the testing feature', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });