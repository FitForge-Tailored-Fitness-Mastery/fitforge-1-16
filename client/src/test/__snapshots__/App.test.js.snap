// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the testing feature 1`] = `
<div
  className="App"
>
  <div
    className="login-container"
  >
    <div
      className="login-box"
    >
      <form
        className="login-form"
        onSubmit={[Function]}
      >
        <h1>
          Login
        </h1>
        <div
          className="input-group"
        >
          <label
            htmlFor="email"
          >
            Username
          </label>
          <input
            id="email"
            onChange={[Function]}
            placeholder="username@email.com"
            type="email"
            value=""
          />
        </div>
        <div
          className="input-group"
        >
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            id="password"
            onChange={[Function]}
            placeholder="********"
            type="password"
            value=""
          />
        </div>
        <button
          className="login-button"
          type="submit"
        >
          Login
        </button>
        <p
          className="signup-link"
        >
          New user? 
          <a
            href="/signup"
            onClick={[Function]}
          >
            Sign up
          </a>
        </p>
      </form>
    </div>
  </div>
</div>
`;
